src/Native/System.Security.Cryptography.Native/pal_ssl.cpp

```diff
@@ -170,35 +170,45 @@ extern "C" int32_t CryptoNative_SslSessionReused(SSL* ssl)
     return SSL_session_reused(ssl) == 1;
 }
 
+static bool StringSpanEquals(const char* lhs, const char* rhs, size_t lhsLength)
+{
+    if (lhsLength != strlen(rhs))
+    {
+        return false;
+    }
+
+    return strncmp(lhs, rhs, lhsLength) == 0;
```

Comment:
> Can't we just use strcmp instead to avoid the branch (and this whole function); since we want to compare the whole string?

But we don't want to compare the whole string, at least not the way I read the code that produces ```encryption``` and ```encryptionLength```.  It's walking an input string until a particular delimiter, which may not always be ```'\0'``` but could also be ```' '```.